src/console_op/console.go

```diff
@@ -0,0 +1,454 @@
+	})
+	if err != nil {
+		log.Printf("doInteractConsole: failed to execute command in pod: %v", err)
+	}
+}
+
+// Handle the console log tail operation.
+func (cs ConsoleManager) doTailConsole(w http.ResponseWriter, r *http.Request) {
+	// This is accessed with a connection that can be upgraded to a websocket.
+
+	// only allow 'GET' calls
+	if r.Method != http.MethodGet {
+		w.Header().Set("Allow", "GET")
+		sendJSONError(w, http.StatusMethodNotAllowed,
+			fmt.Sprintf("(%s) Not Allowed", r.Method))
+		return
+	}
+
+	// Make sure this is a valid operation and all required information is present
+	xname, podName, errMsg, errCode := cs.validateNode(r)
+	if errCode != http.StatusOK {
+		var body = BaseResponse{
+			Msg: errMsg,
+		}
+		SendResponseJSON(w, errCode, body)
+		return
+	}
+
+	// upgrade https to secure websocket connection
+	conn, err := upgrader.Upgrade(w, r, nil)
+	defer conn.Close()
+	if err != nil {
+		fmt.Println("Error upgrading:", err)
+		return
+	}
+
+	// start building the remote command from the input options
+	cmd := []string{"tail"}
+
+	// Find if this is following or just dumping the log
+	if r.Header.Get(consoleHeaderFollowKey) == "True" {
+		// NOTE: use '-F' so the follow works through a log rotation
+		cmd = append(cmd, "-F")
+	}
+
+	// Find if there is a number of lines to display
+	numLines := r.Header.Get(consoleHeaderTailKey)
+	if numLines != "" {
+		cmd = append(cmd, "-n", numLines)
+	}
+
+	// add the filename to the command
+	filename := fmt.Sprintf("/var/log/conman/console.%s", xname)
+	cmd = append(cmd, filename)
+
+	// Set up the remote request to the pod
+	req := cs.k8s.getClientSet().CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(podName).
+		Namespace("services").
+		SubResource("exec").
+		VersionedParams(&v1.PodExecOptions{
+			Command:   cmd,
+			Container: "cray-console-node",
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    false,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+
+	// create the executor to run the command
+	config, err := rest.InClusterConfig()
+	executor, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		log.Printf("failed to create executor: %v", err)
+	}
+
+	// Execute the command piping I/O to the IOStreamer class
+	webIO := &IOStreamer{conn: conn}
+	err = executor.Stream(remotecommand.StreamOptions{
+		Stdin:  webIO,
+		Stdout: webIO,
+		Stderr: nil,
+		Tty:    true,
+	})
+	if err != nil {
+		log.Printf("WEBSOCKET:: failed to execute command in pod: %v", err)
+	}
+}
+
+/////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////
+// Tenant authentication
+/////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////
+
+var serviceName string = "cray-tapms/v1alpha3"
+
+// CASMCMS-9125: Currently when TAPMS bumps this version, it
```

Comment:
Because of this, do you think it may be better to move this into a separate "definitions" file, rather than in the middle of this one? It probably doesn't matter much, I guess. Just seemed odd to run into this kind of global definition in the middle of the file, more than anything else, I suppose.

I do very much appreciate the comment left here, since we ran into that issue in BOS when they changed it out from under us.

Reply:
I debated that, but I wanted to have the vars defined more closely to where they are used.
Since it is a small stand-alone function I figured I would just do it this way.